### PR TITLE
Feat: Add summary display 

### PR DESCRIPTION
**Pull request by sapiderman**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
1. Implements issue #189 

## How did you implement / how did you fix it  
1.  Adds the --summary flag
2. On Monika start a `monika.pid` file is saved with some json data such as proccess id and start time
3. When summary is run, it will look at `monika.pid` and get some stats off that.
4. edit: addition, monika.pid is removed when monka exits.

## How to test  
1. run monika
2. while monika is still running, start another terminal and type `bin\run --summary` (you have to be on the same cwd)

## Screenshot
image

### CODE REVIEW

File: src/jobs/summary-notification.ts [hunk 1]
```diff
@@ -63,3 +63,126 @@ Notifications: ${summary.numberOfSentNotifications}`,
...
+ * getDaysHours breaks down the date into days  hours minute string
+ * @param {Date} startTime is the start time
+ * @returns {string} text of the date to print
+ */
+function getDaysHours(startTime: Date): string {
+  let duration = Math.abs(
+    (new Date().getTime() - new Date(startTime).getTime()) / 1000
+  )
+  const numDays = Math.floor(duration / 86400)
+  duration -= numDays * 86400 // get the remaining hourse
```
Comment (gsukanto) [inline src/jobs/summary-notification.ts:138]: `hours`

**Review by gsukanto — APPROVED** [review]

File: src/jobs/summary-notification.ts (hunk 1 above)
Comment (sapiderman) [reply src/jobs/summary-notification.ts:143]: oops. thanks!

File: src/jobs/summary-notification.ts [hunk 2]
```diff
@@ -68,3 +69,126 @@ Version: ${userAgent}`,
...
+    }
+  })
+}
+
+// do somee cleanups on exit
+// eventEmitter.on(events.application.terminated, async () => {
+export function cleanPidFile() {
+  fs.unlink('monika.pid', (err) => {
+    if (err) {
+      log.indo('trying to cleanup monika.pid, got err: ', err)
```
Comment (armedi) [inline src/jobs/summary-notification.ts:128]: `log.info`?

File: src/index.ts [hunk 3]
```diff
@@ -159,7 +166,12 @@ class Monika extends Command {
     }),
 
     force: flags.boolean({
-      description: 'force command',
+      description: 'Force commands with a yes whenever Y/n is prompted.',
+      default: false,
+    }),
+
+    summary: flags.boolean({
+      description: 'Display a summay of monika running stats',
```
Comment (armedi) [inline src/index.ts:174]:
```suggestion
      description: 'Display a summary of monika running stats',
```

**Comment by armedi** [comment]
Monika does not quit after displaying stats. Is that expected?

**Comment by sapiderman** [comment]
Yeah.. Summary tried to load up the same sequence to get the counts etc.. And it triggers the checking processes that we don't want. Refactoring. Thanks!

**Review by nicnocquee — CHANGES_REQUESTED** [review]
Adding to @armedi 's comment, `--summary` also doesn't terminate when error occurs:

```
INFO: Could not find the file: monika.pid. Monika is probably not running or ran from a diffent directory
ERROR: Summary notification: ENOENT: no such file or directory, open 'monika.pid'
```

It should exit after showing the error.

File: src/jobs/summary-notification.ts (hunk 2 above)
Comment (sapiderman) [reply src/jobs/summary-notification.ts:140]: oh fixed. thanks!

File: src/index.ts (hunk 3 above)
Comment (sapiderman) [reply src/index.ts:175]: fixed. thanks

**Comment by sapiderman** [comment]
> Adding to @armedi 's comment, `--summary` also doesn't terminate when error occurs:
> 
> ```
> INFO: Could not find the file: monika.pid. Monika is probably not running or ran from a diffent directory
> ERROR: Summary notification: ENOENT: no such file or directory, open 'monika.pid'
> ```
> 
> It should exit after showing the error.

fixed here
image

**Comment by sapiderman** [comment]
latest iteration looks like this:

image

**Review by armedi — APPROVED** [review]

**Review by dennypradipta — APPROVED** [review]

**Review by nicnocquee — APPROVED** [review]

**Pull request merged**